src/solvers/refinement/refined_string_type.h

```diff
@@ -0,0 +1,87 @@
+/********************************************************************\
+
+Module: Type for string expressions used by the string solver.
+        These string expressions contains a field `length`, of type
+        `index_type`, a field `content` of type `content_type`.
+        This module also defines function to recognise the C and java
+        string types.
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVERS_REFINEMENT_REFINED_STRING_TYPE_H
+#define CPROVER_SOLVERS_REFINEMENT_REFINED_STRING_TYPE_H
+
+#include <util/std_types.h>
+#include <util/std_expr.h>
+#include <util/arith_tools.h>
+
+#define STRING_SOLVER_INDEX_WIDTH 32
+#define STRING_SOLVER_C_CHAR_WIDTH 8
+#define STRING_SOLVER_JAVA_CHAR_WIDTH 16
+
+// Internal type used for string refinement
+class refined_string_typet: public struct_typet
+{
+public:
+  explicit refined_string_typet(unsignedbv_typet char_type);
+
+  // Type for the content (list of characters) of a string
+  inline array_typet get_content_type()
+  { return to_array_type((to_struct_type(*this)).components()[1].type());}
```

Comment:
You should not need to do to_struct_type here (I think); the return type should be a const reference.